### PR TITLE
Add aten.min.dim to linalg lowering

### PR DESCRIPTION
**Pull request by frederik-h**

### CODE REVIEW

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/reduction.py [hunk 1]
```diff
@@ -335,6 +335,117 @@ def ReduceMaxAlongDim_basic(module, tu: TestUtils):
...
+    ])
+    def forward(self, a):
+        return torch.ops.aten.min(a, 1, keepdim=True)
+
+@register_test_case(module_factory=lambda: ReduceMinKeepDimReturnBoth())
+def ReduceMinKeepDimReturnBoth_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 4, 5, low=-10, high=-5))
+
+# ==============================================================================
+
```
Comment (frederik-h) [inline projects/pt1/python/torch_mlir_e2e_test/test_suite/reduction.py:448]: The test cases have all been adapted from corresponding aten.max test cases.

File: projects/pt1/e2e_testing/xfail_sets.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
     "Conv2dWithPaddingDilationStrideStaticModule_depthwise_multiplier",
     "IscloseStaticModule_basic",
     "IscloseStaticModuleTrue_basic",
+    "ReduceMinAlongDimUnsignedInt_basic",
```
[the remark this answers is not in the document]
Comment (ramiro050) [reply projects/pt1/e2e_testing/xfail_sets.py:22]: What's the error you're getting? I only see the `AtenMaxDim` test xfailed for the Torchdynamo backend, not the linalg backend, so I would expect the `AtenMinDim` test to work for the linalg backend.

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Thanks! I appreciate the thorough updates of comments and variable names.

File: lib/Conversion/TorchToLinalg/Reduction.cpp [hunk 3]
```diff
@@ -30,70 +30,83 @@ using namespace mlir::torch;
 using namespace mlir::torch::Torch;
 
 namespace {
-// Aten maxdim lowering represents the MaxDim op as an linalg.indexed_generic
-// op, producing two output buffers.
+// Helper for ConvertAtenMinMaxDimOp
+template <typename OpTy> constexpr bool isMaxDimOp();
+template <> constexpr bool isMaxDimOp<AtenMaxDimOp>() { return true; }
+template <> constexpr bool isMaxDimOp<AtenMinDimOp>() { return false; }
```
[the remark this answers is not in the document]
Comment (frederik-h) [reply lib/Conversion/TorchToLinalg/Reduction.cpp]: Right! I have changed it now.

**Review by ramiro050 — APPROVED** [review]
Thanks!

File: projects/pt1/e2e_testing/xfail_sets.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ramiro050) [reply projects/pt1/e2e_testing/xfail_sets.py]: Ah, good catch. Yeah, having the XFAIL is fine then.

**Comment by frederik-h** [comment]
> Thanks!

@ramiro050 Thanks for your helpful review!

**Comment by frederik-h** [comment]
> > Thanks!
> 
> @ramiro050 Thanks for your helpful review!

@ramiro050 @vivekkhandelwal1 Just for my understanding: Is one approving review enough and someone just needs to merge this (I cannot) or are we waiting for another review?

**Comment by stellaraccident** [comment]
Ramiro approving such a patch is certainly enough. Having a quick look and will merge.

**Pull request merged**